sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -129,7 +137,8 @@ case class AtomicCreateTableAsSelectExec(
     }
     val stagedTable = catalog.stageCreate(
       ident, query.schema, partitioning.toArray, properties.asJava)
-    writeToStagedTable(stagedTable, writeOptions, ident)
+    val mode = if (ifNotExists) SaveMode.Ignore else SaveMode.ErrorIfExists
```

Comment:
Same problem here. I think mode should always be Append because that's the only one with reliable behavior that matches v2.